python/ray/autoscaler/v2/instance_manager/reconciler.py

```diff
@@ -1472,21 +1472,36 @@ def _handle_extra_cloud_instances(
                 the cloud provider.
             ray_nodes: The ray cluster's states of ray nodes.
         """
+        Reconciler._handle_extra_cloud_instances_from_cloud_provider(
```

Comment:
We had to handle extra from cloud provider before ray for KubeRay: KubeRay is not setting the RAY_NODE_TYPE_NAME env var yet, so ray nodes are missing instance type info, which is required for us to transition an IM instance to ALLOCATED. 

This is ok for now since users could set it in the yaml, but i think KubeRay should probably do it since it's part of the contract between Ray and cloud providers.

Reply:
File a Kuberay issue?

Reply:
https://github.com/ray-project/kuberay/issues/1965